init.lua

```diff
@@ -45,6 +45,7 @@ boost_cart.cart = {
 	punched = false, -- used to re-send velocity and position
 	velocity = {x=0, y=0, z=0}, -- only used on punch
 	old_dir = {x=0, y=0, z=0},
+	last_nonzero_dir = {x=1, y=0, z=0},
```

Comment:
Did you try to extend `old_dir` instead of adding a new variable? Should be possible IMO.